lms/djangoapps/discussion_api/api.py

```diff
@@ -244,6 +254,148 @@ def get_sorted_modules(category):
     }
 
 
+def _get_user_profile_dict(request, usernames):
+    """
+    Gets user profile details for a list of usernames and creates a dictionary with
+    profile details against username.
+
+    Parameters:
+
+        request: The django request object.
+        usernames: A string of comma separated usernames.
+
+    Returns:
+
+        A dict with username as key and user profile details as value.
+    """
+    username_profile_dict = {}
+    try:
+        request.GET = request.GET.copy()
```

Comment:
What does this line do?  At first glance, it looks like the copy is to keep from modifying the original dict, but then it gets assigned immediately back to request.GET, which discards the original.  Am I missing something?